### ISSUE TITLE
MySQL can't parse triggers with complex statements

### ISSUE THREAD

**Issue by andrewmcfaul** [k0kubun/sqldef#203]
When comparing two schemas that contain trigger definitions with subqueries that contain brackets the parsing fails with an error

## Input SQL

```mysql
CREATE TABLE `test_trigger` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `set_id` int(11) NOT NULL,
  `sort_order` int(11) NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;

CREATE TRIGGER test_trigger_BEFORE_INSERT BEFORE INSERT ON test_trigger FOR EACH ROW BEGIN
        IF NEW.sort_order IS NULL OR NEW.sort_order = 0 THEN
                SET NEW.sort_order = (SELECT COALESCE(MAX(sort_order) + 1, 1) FROM test_trigger WHERE set_id = NEW.set_id);
        END IF;
END;
```

## Current output
found syntax error when parsing DDL
```mysql
CREATE TRIGGER test_trigger_BEFORE_INSERT BEFORE INSERT ON test_trigger FOR EACH ROW BEGIN
        IF NEW.sort_order IS NULL OR NEW.sort_order = 0 THEN
                SET NEW.sort_order = (SELECT COALESCE(MAX(sort_order) + 1, 1) FROM test_trigger WHERE set_id = NEW.set_id);
        END IF;
END;
```

## Expected output
It should parse the trigger definitions

**Comment by k0kubun** [comment]
Thanks for your report. Do you have any other examples that should be fixed? Is that the only trigger that you have and is incompatible with the sqldef parser?

**Comment by andrewmcfaul** [comment]
Hi, I believe it is just this trigger. If I remove it and run the script again it parses everything else correctly

**Comment by k0kubun** [comment]
Fixed the reported test case in v0.11.36.
